test/integration/extension_discovery_integration_test.cc

```diff
@@ -323,5 +327,17 @@ TEST_P(ExtensionDiscoveryIntegrationTest, BasicTwoSubscriptionsSameName) {
   EXPECT_EQ("200", response->headers().getStatusValue());
 }
 
+TEST_P(ExtensionDiscoveryIntegrationTest, DestroyDuringInit) {
```

Comment:
Can you add more comments in here? It's not clear to me how rate limiting is part of this and what is going on here exactly. 

Also, like I said before, a unit test is fine if it's easier to test this case deterministically. Up to you.

Thanks a ton for working on this.

/wait

Reply:
I have no idea why rate limiting is important, but it allows me to get through to the offending line here: https://github.com/envoyproxy/envoy/blob/master/source/common/config/grpc_mux_impl.cc#L258. The block prior to that is guarded by rate limiting check.

Reply:
OK this feels pretty fragile to me, but I would like to merge this to unflake things. If we can figure out a better way to test this vs. this as a follow up that would be good. cc @lizan who might have additional thoughts.

Reply:
Yeah, I don't think stat update should be guarded by an exceptional event (rate limit kicking in). It creates a really tough edge case to hit.